### PR TITLE
codeintel: Cleanup GraphQL utilities

### PR DESCRIPTION
**Pull request by efritz**
Cleanup and consolidate GraphQL utils in codeintel.

## Test plan

Relying on CI here.

### CODE REVIEW

File: enterprise/internal/codeintel/uploads/transport/graphql/root_resolver.go [hunk 1]
```diff
@@ -6,51 +6,35 @@ import (
...
-	siteAdminChecker        sharedresolvers.SiteAdminChecker
-	repoStore               database.RepoStore
-	prefetcherFactory       *sharedresolvers.PrefetcherFactory
-	locationResolverFactory *sharedresolvers.CachedLocationResolverFactory
+	uploadSvc  UploadService
+	operations *operations
 }
 
-func NewRootResolver(observationCtx *observation.Context, uploadSvc UploadService, policySvc PolicyService, gitserverClient gitserver.Client, siteAdminChecker sharedresolvers.SiteAdminChecker, repoStore database.RepoStore, prefetcherFactory *sharedresolvers.PrefetcherFactory, locationResolverFactory *sharedresolvers.CachedLocationResolverFactory) resolverstubs.UploadsServiceResolver {
+func NewRootResolver(observationCtx *observation.Context, uploadSvc UploadService) resolverstubs.UploadsServiceResolver {
```
Comment (Numbers88s) [inline enterprise/internal/codeintel/uploads/transport/graphql/root_resolver.go:18]: Fire 🔥 how come you were able to remove so much from the root resolver? Is it within the service?

Comment (efritz) [reply enterprise/internal/codeintel/uploads/transport/graphql/root_resolver.go:18]: Uploads literally only have one resolver and it's commit graph (not even directly accessible). Would like *next* to concentrate on moving a bunch of the metadata/record/visibility stuff from autoindexing back into uploads, possibly rename it, then leave autoindexing service as the thing that does scheduling/inference/executor interfacing **only**.

Comment (Numbers88s) [reply enterprise/internal/codeintel/uploads/transport/graphql/root_resolver.go:18]: Gotcha gotcha, yeah you told me about that. I'm just surprised that you were able to move all that functionality i.e. 
```
        autoindexSvc            AutoIndexingService
	policySvc               PolicyService
	gitserverClient         gitserver.Client
	operations              *operations
	siteAdminChecker        sharedresolvers.SiteAdminChecker
	repoStore               database.RepoStore
	prefetcherFactory       *sharedresolvers.PrefetcherFactory
	locationResolverFactory *sharedresolvers.CachedLocationResolverFactory
```

Did all that get moved to another layer or service or was it simply not used? Or was it unnecessary?

Comment (efritz) [reply enterprise/internal/codeintel/uploads/transport/graphql/root_resolver.go:18]: It was all stuff passed to `sharedresolvers` at some point.

Comment (Numbers88s) [reply enterprise/internal/codeintel/uploads/transport/graphql/root_resolver.go:18]: Wild. 

Well the more we can do this and pass all logic to the services the better/cleaner it'll be. So +1 to remove as much as possible from transport/resolver layer.

**Review by Numbers88s — APPROVED** [review]

**Pull request merged**